examples/lock-app/cc13x2x7_26x2x7/main/AppTask.cpp

```diff
@@ -151,6 +151,27 @@ int AppTask::Init()
             ;
     }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING && CHIP_ENABLE_ROTATING_DEVICE_ID
+    {
+	uint8_t EUI64[8];
+        char stringEUI64[sizeof(EUI64) * 2 + 1]; // just big enough for string(EUI64)
+
+        // set serial number to flash
+        // this is before ZCL is started, so Basic Cluster populates Serial Number with this EUI64 (in string)
+        ThreadStackMgrImpl().GetIeeeEui64(EUI64);
+        sprintf(stringEUI64, "%02X%02X%02X%02X%02X%02X%02X%02X",
+			EUI64[0], EUI64[1], EUI64[2], EUI64[3], EUI64[4], EUI64[5], EUI64[6], EUI64[7]);
+	ret = ConfigurationMgr().StoreSerialNumber(stringEUI64, strlen(stringEUI64));
+        PLAT_LOG("StoreSerialNumber: %s", stringEUI64);
+	if (ret != CHIP_NO_ERROR)
+        {
+            PLAT_LOG("ConfigurationMgr().StoreSerialNumber failed (%X)", ret.AsInteger());
+            while (1)
+                ;
```

Comment:
Why the busy-loop?

Reply:
Maybe ChipDie would be more approrpiate. Or replace the if with `VerifyOrDie`

Reply:
Thanks for the suggestions, those sound like good candidates which would have similar effect as while loop, but much more intentional. This may also apply to our other busy-loops in this file.

Reply:
Updated here: https://github.com/project-chip/connectedhomeip/pull/15367/commits/9bd267a4b72636f48a7990b51968bb575aaa1a01